src/DIRAC/WorkloadManagementSystem/Agent/CloudDirector.py

```diff
@@ -0,0 +1,527 @@
+          # if the vmType definition did not change
+          vmTypeHash = self.__generateVMTypeHash(ceVMTypeDict)
+          if vmTypeName in self.vmTypeCECache and self.vmTypeCECache[vmTypeName]['Hash'] == vmTypeHash:
+            vmTypeCE = self.vmTypeCECache[vmTypeName]['CE']
+          else:
+            result = ceFactory.getCEObject(parameters=ceVMTypeDict)
+            if not result['OK']:
+              return result
+            self.vmTypeCECache.setdefault(vmTypeName, {})
+            self.vmTypeCECache[vmTypeName]['Hash'] = vmTypeHash
+            self.vmTypeCECache[vmTypeName]['CE'] = result['Value']
+            vmTypeCE = self.vmTypeCECache[vmTypeName]['CE']
+            vmTypeCE.setBootstrapParameters(opParameters)
+
+          self.vmTypeDict[vmTypeName]['CE'] = vmTypeCE
+          self.vmTypeDict[vmTypeName]['CEName'] = ce
+          self.vmTypeDict[vmTypeName]['CEType'] = ceDict['CEType']
+          self.vmTypeDict[vmTypeName]['Site'] = site
+          self.vmTypeDict[vmTypeName]['VMType'] = vmType
+          self.vmTypeDict[vmTypeName]['Platform'] = platform
+          self.vmTypeDict[vmTypeName]['MaxInstances'] = ceDict['MaxInstances']
+          if not self.vmTypeDict[vmTypeName]['CE'].isValid():
+            self.log.error('Failed to instantiate CloudEndpoint for %s' % vmTypeName)
+            continue
+
+          if site not in self.sites:
+            self.sites.append(site)
+
+    return S_OK()
+
+  def execute(self):
+    """ Main execution method
+    """
+
+    if not self.vmTypeDict:
+      self.log.warn('No site defined, exiting the cycle')
+      return S_OK()
+
+    result = self.createVMs()
+    if not result['OK']:
+      self.log.error('Errors in the job submission: ', result['Message'])
+
+    # cyclesDone = self.am_getModuleParam( 'cyclesDone' )
+    # if self.updateStatus and cyclesDone % self.cloudStatusUpdateCycleFactor == 0:
+    #  result = self.updatePilotStatus()
+    #  if not result['OK']:
+    #    self.log.error( 'Errors in updating cloud status: ', result['Message'] )
+
+    return S_OK()
+
+  def createVMs(self):
+    """ Go through defined computing elements and submit jobs if necessary
+    """
+
+    vmTypeList = self.vmTypeDict.keys()
+
+    # Check that there is some work at all
+    setup = CSGlobals.getSetup()
+    tqDict = {'Setup': setup,
+              'CPUTime': 9999999}
+    if self.vo:
+      tqDict['VO'] = self.vo
+    if self.voGroups:
+      tqDict['OwnerGroup'] = self.voGroups
+
+    result = Resources.getCompatiblePlatforms(self.platforms)
+    if not result['OK']:
+      return result
+    tqDict['Platform'] = result['Value']
+    tqDict['Site'] = self.sites
+    tags = []
+    for vmType in vmTypeList:
+      if 'Tag' in self.vmTypeDict[vmType]['ParametersDict']:
+        tags += self.vmTypeDict[vmType]['ParametersDict']['Tag']
+    tqDict['Tag'] = list(set(tags))
+    tqDict['SubmitPool'] = "wenmrPool"
```

Comment:
really?

Reply:
I've removed the SubmitPool entirely, it clearly couldn't have been doing anything useful.